pandas/_libs/groupby.pyx

```diff
@@ -1075,6 +1075,13 @@ def group_mean(
                     y = val - compensation[lab, j]
                     t = sumx[lab, j] + y
                     compensation[lab, j] = t - sumx[lab, j] - y
+                    if compensation[lab, j] != compensation[lab, j]:
+                        # GH#50367
+                        # if val is +/- infinity, this (t-sumx[lab, j]-y) expression
+                        # will evaluate to Nan. So we are checking whether
+                        # compensation[lab, j] is Nan and setting it to 0.
+                        # we cannot use util.is_nan because of no gil
```

Comment:
I think this could focus a little more on why. What do you think of something like

```
# If val is +/- infinity, compensation is NaN which would lead
# to results being NaN instead of +/-infinity.
# We cannot use util.is_nan because of no gil
```

Reply:
Done @rhshadrach

Reply:
Hi @rhshadrach .. Have addressed your comment.. can you pls resolve the change request ?